redaxo/src/core/lib/response.php

```diff
@@ -158,6 +159,28 @@ public static function sendFile($file, $contentType, $contentDisposition = 'inli
         self::sendAdditionalHeaders();
         self::sendPreloadHeaders();
 
+        if (PHP_VERSION_ID >= 50600) {
+            header('Accept-Ranges: bytes');
+            $rangeHeader = rex_request::server('HTTP_RANGE', 'string', null);
+            if ($rangeHeader) {
+                $unitFactory = new \Ramsey\Http\Range\UnitFactory();
+                $ranges = $unitFactory->getUnit(trim($rangeHeader), filesize($file))->getRanges();
+                foreach ($ranges as $range) {
+                    header('HTTP/1.1 ' . self::HTTP_PARTIAL_CONTENT);
+                    header('Content-Length: '.$range->getLength());
+                    header('Content-Range: bytes '.$range->getStart().'-'.$range->getEnd().'/'.filesize($file));
+
+                    $handle = fopen($file, 'rb');
```

Comment:
muss man das handle wirklich immer wieder öffnen innerhalb der schleife?

Reply:
In der readme der lib ist folgendes beispiel:

```
file_get_contents(
            $filePath,
            false,
            null,
            $rangeValue->getStart(),
            $rangeValue->getLength()
        );
```

Reply:
ich hatte diese variante nicht gewählt, da wenn ich alles auch einfach einlese, ich ein memory problem bekomme. Ist die Frage, ob x-mal file_get_contents besser ist, als mit file-handle zu arbeiten.

Reply:
ein filehandle brauch "keinen php speicher", file_get_contents() brauch so viel kb speicher im php, wie groß die datei ist

Reply:
d.h. die variante mit handle sollte effektiver sein